### PR TITLE
Quiet tests.

### PR DESCRIPTION
**Pull request by dweindl**
Don't print all the test output by default, only on failures.

### CODE REVIEW

File: test/base/test_prior.py [hunk 1]
```diff
@@ -125,7 +125,7 @@ def test_mode(scale, prior_type_list):
 
             # flat functions don't have local minima, so dont check this
             # for uniform priors
-
```
Comment (PaulJonasJost) [inline test/base/test_prior.py:128]: Is this an actual output we want? especially since it is a print?

File: tox.ini [hunk 2]
```diff
@@ -110,15 +109,15 @@ deps =
...
-        test/hierarchical -s
+        test/hierarchical
 description =
     Test hierarchical optimization module
 
 [testenv:select]
 extras = test,amici,petab,select
 commands =
     pytest --cov=pypesto --cov-report=xml --cov-append \
-        test/select -s
```
Comment (PaulJonasJost) [inline tox.ini:121]: not sure but don't we become "less silent" by setting the capture method from `no` to `fd`?

**Review by PaulJonasJost — APPROVED** [review]

File: test/base/test_prior.py (hunk 1 above)
Comment (dweindl) [reply test/base/test_prior.py:128]: Ooops, definitely not.

File: test/base/test_prior.py [hunk 3]
```diff
@@ -125,7 +125,7 @@ def test_mode(scale, prior_type_list):
...
+            print("XXXXX", result.optimize_result.list[0].fval)
```
Comment (dweindl) [inline test/base/test_prior.py:128]:
```suggestion
```

File: tox.ini (hunk 2 above)
Comment (dweindl) [reply tox.ini:121]: Compare:
* now: https://github.com/ICB-DCM/pyPESTO/actions/runs/6912941885/job/18809255758
* before: https://github.com/ICB-DCM/pyPESTO/actions/runs/6905429814/job/18788171857

**Pull request merged**